primeqa/components/reranker/colbert_reranker.py

```diff
@@ -0,0 +1,161 @@
+
+    Returns:
+        _type_: _description_
+
+    """
+
+    '''checkpoint: str = field(
+        default=None,
+        metadata={
+            "name": "Checkpoint",
+            "description": "Path to checkpoint",
+        },
+    )
+    '''
+
+    model: str = field(
+        default="drdecr", #"cross-encoder/ms-marco-TinyBERT-L-2",
+        metadata={
+            "name": "Model",
+            "api_support": True,
+            "description": "Path to model",
+        },
+    )
+
+    max_num_documents: int = field(
+        default=5,
+        metadata={
+            "name": "Maximum number of retrieved documents",
+            "range": [1, 100, 1],
+            "api_support": True,
+            "exclude_from_hash": True,
+        },
+    )
+
+    include_title: bool = field(
+        default=True,
+        metadata={
+            "name": "Include Title",
+            "description": "Whether to concate text and title",
+            "choices": "True|False"
+        },
+    )
+
+    def __post_init__(self):
+        self._config = ColBERTConfig(
+            index_root=None,
+            index_name=None,
+            index_path=None,
+        )
+
+        # Placeholder variables
+        self._searcher = None
+
+    def __hash__(self) -> int:
+        # Step 1: Identify all fields to be included in the hash
+        hashable_fields = [
+            k
+            for k, v in self.__class__.__dataclass_fields__.items()
+            if not "exclude_from_hash" in v.metadata
+            or not v.metadata["exclude_from_hash"]
+        ]
+
+        # Step 2: Run
+        return hash(
+            f"{self.__class__.__name__}::{json.dumps({k: v for k, v in vars(self).items() if k in hashable_fields}, sort_keys=True)}"
+        )
+
+    def load(self, *args, **kwargs):
+        self._loaded_model = Searcher(
+            None,
+            checkpoint=self.model,
+            collection=None,
+            config=self._config,
+            rescore_only=True
+        )
+
+    def train(self, *args, **kwargs):
+        pass
+
+    def eval(self, *args, **kwargs):
+        pass
+
+    def predict(self, queries: List[str],
+                    documents:  List[List[Dict]],
+                    *args,
+                    **kwargs):
+        """
+        Args:
+            queries (List[str]): search queries
+            texts (List[List[Dict]]): For each query, a list of documents to rerank
+                where each document is a dictionary with the following structure:
+                {
+                    "document": {
+                        "text": "A man is eating food.",
+                        "document_id": "0",
+                        "title": "food"
+                    },
+                    "score": 1.4
+                }
+
+        Returns:
+            List[List[Dict]] For each query a list of reranked documents in the same
+            structure as the input documents with the score replace with the reranker score.
```

Comment:
Suggested replacement for the two lines: 

List[List[Dict]] A list of reranked documents in the same structure as the input documents with the score replaced with the reranker score for each query.

Reply:
Updated as suggested